pkg/controller/spinnakervalidating/spinnakervalidating_controller.go

```diff
@@ -23,66 +18,35 @@ import (
 // spinnakerValidatingController annotates Pods
 type spinnakerValidatingController struct {
 	client  client.Client
-	decoder types.Decoder
+	decoder admission.Decoder
 }
 
 // NewSpinnakerService instantiates the type we're going to validate
-var SpinnakerServiceBuilder v1alpha1.SpinnakerServiceBuilderInterface
+var SpinnakerServiceBuilder v1alpha2.SpinnakerServiceBuilderInterface
 
 // Implement admission.Handler so the controller can handle admission request.
 var _ admission.Handler = &spinnakerValidatingController{}
 var log = logf.Log.WithName("spinvalidate")
 
 // Add adds the validating admission controller
 func Add(m manager.Manager) error {
-	ns, err := k8sutil.GetOperatorNamespace()
-	if err != nil {
-		return err
-	}
-
-	validatingWebhook, err := builder.NewWebhookBuilder().
-		Name("validating.k8s.io").
-		Validating().
-		Operations(admissionregistrationv1beta1.Create, admissionregistrationv1beta1.Update).
-		WithManager(m).
-		ForType(SpinnakerServiceBuilder.New()).
-		Handlers(&spinnakerValidatingController{}).
-		Build()
-
+	//ns, err := k8sutil.GetOperatorNamespace()
+	_, err := k8sutil.GetOperatorNamespace()
 	if err != nil {
 		return err
 	}
 
-	disableWebhookConfigInstaller := false
-
-	as, err := webhook.NewServer("spinnaker-admission-server", m, webhook.ServerOptions{
-		Port:                          9876,
-		CertDir:                       "/tmp/cert",
-		DisableWebhookConfigInstaller: &disableWebhookConfigInstaller,
-		BootstrapOptions: &webhook.BootstrapOptions{
-			Service: &webhook.Service{
-				Namespace: ns,
-				Name:      "spinnaker-admission-service",
-				// Selectors should select the pods that runs this webhook server.
-				Selectors: map[string]string{
-					"name": "spinnaker-operator",
-				},
-			},
-		},
-	})
-
-	if err != nil {
-		return err
-	}
-	return as.Register(validatingWebhook)
+	hookServer := m.GetWebhookServer()
+	hookServer.Register("/validate-v1alpha1-spinnakerservice", &webhook.Admission{Handler: &spinnakerValidatingController{}})
```

Comment:
Where are you specifying for which `kinds` is the webhook invoked? Also, how is the kubernetes `Service` created? It was created by the webhook BootstrapOptions.